### PR TITLE
Setup/husky

### PR DESCRIPTION
**Pull request by AniaPiWo**
npx husky-init && npm install

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Setup pre-commit hook using husky. It will run our maintenance tools before committing changes.

### Requirements
1. When doing commit, Husky pre-commit hook should run `yarn lint`, and `yarn pretty` scripts.
2. Husky should run pre-commit check for commit messages. We should use [this library](https://github.com/conventional-changelog/commitlint)

### Review

#### .husky/pre-commit
The pre-commit hook is set to run `npm test` twice, which does not align with the task requirements. According to the requirements, the pre-commit hook should run `yarn lint` and `yarn pretty`. The code should be modified as follows:

```diff
#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

- npm test
- npm test
+ yarn lint
+ yarn pretty
```

#### package.json
The husky package has been correctly added to the devDependencies, and the `prepare` script has been added to install husky. However, there is no setup for commitlint as required in the task. 

To fulfill the requirements, you need to add commitlint to your devDependencies and configure it in your package.json file. Here is an example of how you can do this:

```diff
"devDependencies": {
  ...
  "husky": "^8.0.0",
+ "commitlint": "^15.0.0",
+ "@commitlint/config-conventional": "^15.0.0"
},
"husky": {
+ "hooks": {
+   "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
+ }
},
"commitlint": {
+ "extends": ["@commitlint/config-conventional"]
}
```

Please note that you need to install the `commitlint` and `@commitlint/config-conventional` packages for this setup to work. You can do this by running `yarn add --dev commitlint @commitlint/config-conventional`.

In conclusion, the provided code changes do not fully cover the task requirements. The pre-commit hook needs to be updated to run the correct scripts, and commitlint needs to be set up for commit message checks.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
# Code Review

## Task Requirements Validation

1. The task requires setting up a Husky pre-commit hook to run `yarn lint` and `yarn pretty` scripts. However, the provided changes in `.husky/pre-commit` file do not meet this requirement. The pre-commit hook is currently set to run `npm test` instead of the required scripts.

2. The task also requires Husky to run a pre-commit check for commit messages using the commitlint library. This requirement is partially met. The `package.json` file shows that Husky, commitlint, and @commitlint/config-conventional have been added as devDependencies, and a commit-msg hook has been set up to use commitlint with the conventional config. However, the commitlint configuration should be in a separate file named `commitlint.config.js` or `.commitlintrc.js`, not in the `package.json` file.

## Suggested Changes

1. Update the `.husky/pre-commit` file to run the required scripts:

```diff
#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

-npm test
+yarn lint
+yarn pretty
```

2. Move the commitlint configuration to a separate file named `commitlint.config.js`:

```javascript
module.exports = {
  extends: ['@commitlint/config-conventional']
};
```

And remove it from the `package.json`:

```diff
"husky": {
  "hooks": {
    "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
  }
},
-"commitlint": {
-  "extends": [
-    "@commitlint/config-conventional"
-  ]
-}
```

Please make these changes and resubmit for review.
